mongo/mongo_test.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package mongo
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/flamego/flamego"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/flamego/session"
+)
+
+func newTestDB(t *testing.T, ctx context.Context) (testDB *mongo.Database, cleanup func() error) {
+	uri := os.Getenv("MONGODB_URI")
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	if err != nil {
+		t.Fatalf("Failed to connect to mongo: %v", err)
+	}
+
+	dbname := "flamego-test-sessions"
+	err = client.Database(dbname).Drop(ctx)
+	if err != nil {
+		t.Fatalf("Failed to drop test database: %v", err)
+	}
+	db := client.Database(dbname)
+	t.Cleanup(func() {
+		if t.Failed() {
+			t.Logf("DATABASE %s left intact for inspection", dbname)
+			return
+		}
+
+		err = db.Drop(ctx)
+		if err != nil {
+			t.Fatalf("Failed to drop test database: %v", err)
+		}
+	})
+	return db, func() error {
+		if t.Failed() {
+			return nil
+		}
+
+		err = db.Collection("sessions").Drop(ctx)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func TestMongoDBStore(t *testing.T) {
```

Comment:
```suggestion
func TestMongoStore(t *testing.T) {
```

We have been calling it `MongoStore` in `mongo.go`.